### PR TITLE
fix: validate uuid loaded from disk (#68)

### LINKED ISSUE #68: Validate UUID before using it
Given that the UUID can be altered manually in the file (see comment at https://plugins.jetbrains.com/plugin/16209-telemetry-by-red-hat/reviews#review=81120), we should validate the UUID before using it.

ex. as discussed at Baeldung: https://www.baeldung.com/java-validate-uuid-string#hexagonal

### PR DESCRIPTION
**Pull request by adietish**
fixes #68

### CODE REVIEW

File: src/main/java/com/redhat/devtools/intellij/telemetry/core/service/UserId.java [hunk 1]
```diff
@@ -60,12 +70,26 @@ private String load(Path uuidFile) {
...
+    private String create() {
+        return UUID.randomUUID().toString();
+    }
+
+    /** for testing purposes */
+    protected String write(String uuid, Path uuidFile) {
         try {
             FileUtils.createFileAndParent(uuidFile);
             FileUtils.write(uuid, uuidFile);
+            return uuid;
```
Comment (sbouchet) [inline src/main/java/com/redhat/devtools/intellij/telemetry/core/service/UserId.java:89]: why return the passed uuid here ? 
doesn't make difference to keep same signature as before.

File: src/main/java/com/redhat/devtools/intellij/telemetry/core/service/UserId.java [hunk 2]
```diff
@@ -28,26 +28,41 @@ public class UserId {
...
         } else {
-            String uuid = UUID.randomUUID().toString();
+            uuid = create();
             write(uuid, file);
-            return uuid;
         }
+        return uuid;
+    }
+        }
+    }
```
Comment (sbouchet) [inline src/main/java/com/redhat/devtools/intellij/telemetry/core/service/UserId.java:57]: those seems extra 
```suggestion
    }
```

File: src/main/java/com/redhat/devtools/intellij/telemetry/core/service/UserId.java [hunk 3]
```diff
@@ -28,26 +28,36 @@ public class UserId {
...
-            String uuid = UUID.randomUUID().toString();
-            write(uuid, file);
+        if (exists(file)) {
+            String uuid = load(file);
+            if (!isValid(uuid)) {
+                uuid = write(create(), file);
+            }
             return uuid;
+        } else {
+            return write(create(), file);
```
[the remark this answers is not in the document]
Comment (sbouchet) [reply src/main/java/com/redhat/devtools/intellij/telemetry/core/service/UserId.java]: @fbricon love it !

[the remark this answers is not in the document]
Comment (adietish) [reply src/main/java/com/redhat/devtools/intellij/telemetry/core/service/UserId.java]: incredible, 10 lines of code, as if it was worth 3 people discussing it...

[the remark this answers is not in the document]
Comment (fbricon) [reply src/main/java/com/redhat/devtools/intellij/telemetry/core/service/UserId.java]: It's OK I only spent 8h fine-tuning these 10LOC

[the remark this answers is not in the document]
Comment (adietish) [reply src/main/java/com/redhat/devtools/intellij/telemetry/core/service/UserId.java]: LOL, I needed a 1 months seclusion in a buddhist monastery and it my version didnt even reach consensus

**Pull request merged**